rust/crates/libs/edlcodegen-core/src/edl_core_ffi.rs

```diff
@@ -0,0 +1,63 @@
+#![allow(non_camel_case_types, non_snake_case, unused_imports)]
+use crate::edl_core_types::AbiError;
+use core::ffi::c_void;
+
+// The following subset of types/consts were taken from the windows_sys::Win32::Foundation.
+// We need to make sure that this crate can be usable in both the host and the enclave
+// and not accidentally bring in things that are not usable inside an enclave.
+pub type EnclaveRoutine = isize;
+pub type HANDLE = *mut core::ffi::c_void;
+pub type HEAP_FLAGS = u32;
+pub type WIN32_ERROR = u32;
+pub const HEAP_ZERO_MEMORY: HEAP_FLAGS = 8u32;
+pub const TRUE: windows_sys::core::BOOL = 1i32;
+pub const FALSE: windows_sys::core::BOOL = 0i32;
+pub const S_OK: windows_sys::core::HRESULT = 0x0_u32 as _;
+pub const E_FAIL: windows_sys::core::HRESULT = 0x80004005_u32 as _;
+pub const E_INVALIDARG: windows_sys::core::HRESULT = 0x80070057_u32 as _;
+pub static WIN32_FALSE: windows_result::BOOL = windows_result::BOOL(FALSE);
+pub static WIN32_TRUE: windows_result::BOOL = windows_result::BOOL(TRUE);
+
+// The consuming crate will link these in based on which side of the trust boundary the crate exists
+// in. Either Kernel32 for the host or vertdll for the enclave.
+unsafe extern "system" {
+    pub fn CallEnclave(
```

Comment:
>CallEnclave [](http://example.com/codeflow?start=11&length=11)

How come we can't link in windows-rs impl?

[https://github.com/microsoft/windows-rs/blob/945130accc25ac18a47054115e861ca704a37eb5/crates/libs/windows/src/Windows/Win32/System/Environment/mod.rs#L2](https://github.com/microsoft/windows-rs/blob/945130accc25ac18a47054115e861ca704a37eb5/crates/libs/windows/src/Windows/Win32/System/Environment/mod.rs#L2)